### DISCUSSION TITLE
What is the difference between the new extension and the classic one?

### DISCUSSION THREAD

**Discussion by Skyxim** [Tampermonkey/tampermonkey#2255] [Q&A]
The new one supports more Apple devices, but does it have any advantages or disadvantages compared to the classic one in terms of macOS experience?

**Comment by derjanb** [comment]
Tampermonkey Classic is based on Safari App Extension technology, which was the only available way to create extensions for Safari in 2019. App extensions are not really designed for extensions with a background page and therefore many things need to be re-implemented in Swift and a JavaScript wrapper.
This works good in general, but the extension sometimes gets confused with tab and window IDs. Also GM_xmlhttpRequest does not send system cookies, because there is no way to retrieve them.

The new Tampermonkey is based on Safari Web Extension technology, which is the new way to create extensions for Safari. It is a bit more powerful and has a better API support for things a web extension needs to do.
I also think that Web Extension based extensions are a little bit faster and lightweight in terms of memory and CPU usage, but without having benchmarked this.

So if things are working fine and you are not affected by [any of these issue](https://github.com/Tampermonkey/tampermonkey/issues?q=is%3Aissue+is%3Aopen+label%3Asafariapp), then I'd say keep using Tampermonkey Classic. 
Please note that the new Tampermonkey also has [some known issues](https://github.com/Tampermonkey/tampermonkey/issues?q=is%3Aissue+is%3Aopen+label%3Asafari).
